### PR TITLE
Use gfortran 11.3.0 from repackaged conda tarball for x86_64

### PR DESCRIPTION
**Pull request by isuruf**

### CODE REVIEW

**Comment by isuruf** [comment]
This code is untested

File: gfortran_utils.sh [hunk 1]
```diff
@@ -91,21 +91,48 @@ if [ "$(uname)" == "Darwin" ]; then
     GFORTRAN_PATH="gfortran-8.2-Mojave/gfortran.pkg"
     export GFORTRAN_SHA="$(shasum $GFORTRAN_DMG)"
```
Comment (h-vetinari) [inline gfortran_utils.sh]: These lines (and `GFORTRAN_DMG` above) are now unused.

**Comment by andyfaff** [comment]
@isuruf I've just tried using the changes made in this PR in https://github.com/andyfaff/scipy/pull/38. Unfortunately the compiler doesn't seem to work. I tried to compile a test program and I get:
```
  + gfortran --version
  GNU Fortran (GCC) 11.3.0
  Copyright (C) 2021 Free Software Foundation, Inc.
  This is free software; see the source for copying conditions.  There is NO
  warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
  
  + gfortran tools/wheels/test.f
  ld: library not found for -lm
  collect2: error: ld returned 1 exit status
```
The test run is [here](https://github.com/andyfaff/scipy/actions/runs/3407822428/jobs/5667835565).

**Comment by isuruf** [comment]
For the compiler to work, you need to set SDKROOT like you do for arm64 in https://github.com/scipy/scipy/blob/4b73d5e7150e05ac66baef19c6ddffc5ec4b3748/.github/workflows/wheels.yml#L175

**Comment by rgommers** [comment]
Thanks @isuruf. Using gfortran 11.3.0 from conda-forge sounds great to me. Compiler version wise I'm not aware of any issues, and I've been using it locally on arm64 macOS for a little while.

**Comment by isuruf** [comment]
This is ready. The compilers here were tested by @andyfaff in the scipy CI and they work.

**Comment by andyfaff** [comment]
Note, I used the `x86_64-native` and `arm64-cross`, I haven't tried the other two.

**Comment by mattip** [comment]
Why didn't CI run?

**Comment by isuruf** [comment]
CI hasn't run in a while because the CI here used travis-ci.org and it was shut down ~6 months ago. We can move to travis-ci.com, but you'll run out of credits quickly.

**Comment by mattip** [comment]
@isuruf: I think this should force the MACOSX_DEPLOYMENT_TARGET (in `get_distutils_platform`) to 10_9:
```diff
diff --git a/gfortran_utils.sh b/gfortran_utils.sh
index d2ac55f..8cd7573 100644
--- a/gfortran_utils.sh
+++ b/gfortran_utils.sh
@@ -25,9 +25,8 @@ function get_distutils_platform {
         echo "manylinux1_$plat"
         return
     fi
-    # macOS 32-bit arch is i386
-    [ "$plat" == "i686" ] && plat="i386"
-    local target=$(echo $MACOSX_DEPLOYMENT_TARGET | tr .- _)
+    # The gfortran downloads build for macos 10.9
+    local target="10_9"
     echo "macosx_${target}_${plat}"
 }
 
@@ -51,9 +50,8 @@ function get_distutils_platform_ex {
         echo "manylinux${mb_ml_ver}_${plat}"
         return
     fi
-    # macOS 32-bit arch is i386
-    [ "$plat" == "i686" ] && plat="i386"
-    local target=$(echo $MACOSX_DEPLOYMENT_TARGET | tr .- _)
+    # The gfortran downloads build for macos 10.9
+    local target="10_9"
     echo "macosx_${target}_${plat}"
 }

```

**Comment by isuruf** [comment]
@mattip, we shouldn't force the target for other projects. It should be a per project setting.

**Comment by mattip** [comment]
Maybe then a comment that 10.9 is supported, or a check that the value is at least 10.9?

**Comment by isuruf** [comment]
Yes, checking that the value is at least 10.9 is a good idea.

**Pull request merged**